src/main/java/org/candlepin/async/tasks/AnonymousRegisterAndEntitleJob.java

```diff
@@ -0,0 +1,208 @@
+        }
+
+        /**
+         * Sets the cloud account ID for this job
+         *
+         * @param cloudAccountId the cloud account ID to set for this job
+         * @return a reference to this job config
+         */
+        public AnonymousRegisterJobConfig setCloudAccountId(String cloudAccountId) {
+            if (cloudAccountId == null || cloudAccountId.isEmpty()) {
+                throw new IllegalArgumentException("cloudAccountId is null or empty");
+            }
+
+            this.setJobArgument(ACCOUNT_ID, cloudAccountId);
+
+            return this;
+        }
+
+        /**
+         * Sets the cloud offering ID for this job
+         *
+         * @param cloudOfferingId the cloud offering ID to set for this job
+         * @return a reference to this job config
+         */
+        public AnonymousRegisterJobConfig setCloudOfferingId(String cloudOfferingId) {
+            if (cloudOfferingId == null || cloudOfferingId.isEmpty()) {
+                throw new IllegalArgumentException("cloudOfferingId is null or empty");
+            }
+
+            this.setJobArgument(OFFERING_ID, cloudOfferingId);
+
+            return this;
+        }
+
+        /**
+         * Sets the owner key for this job
+         *
+         * @param ownerKey key of organization which want to entitle offering
+         * @return a reference to this job config
+         */
+        public AnonymousRegisterJobConfig setOwnerKey(String ownerKey) {
+            if (ownerKey == null || ownerKey.isEmpty()) {
+                throw new IllegalArgumentException("ownerKey is null or empty");
+            }
+
+            this.setJobArgument(OWNER_KEY, ownerKey);
+
+            return this;
+        }
+
+        /**
+         * Sets the cloud provider for this job
+         *
+         * @param cloudProviderShortName shortcut of the cloud provider
+         * @return a reference to this job config
+         */
+        public AnonymousRegisterJobConfig setCloudProvider(CloudProvider cloudProviderShortName) {
+            if (cloudProviderShortName == null) {
+                throw new IllegalArgumentException("cloudProviderShortName is null");
+            }
+
+            this.setJobArgument(CLOUD_PROVIDER, cloudProviderShortName);
+
+            return this;
+        }
+
+        @Override
+        public void validate() throws JobConfigValidationException {
+            super.validate();
+
+            try {
+                JobArguments arguments = this.getJobArguments();
+
+                String accountId = arguments.getAsString(ACCOUNT_ID);
+                String offeringId = arguments.getAsString(OFFERING_ID);
+                String cloudProviderShortName = arguments.getAsString(CLOUD_PROVIDER);
+
+                if (accountId == null || accountId.isEmpty()) {
+                    String errmsg = "Account has not been set, or the provided account is empty";
```

Comment:
Let's be more specific: `Cloud Account ID has not been set, or is empty`

Reply:
done